packages/web-components/src/components/cta-block/__stories__/cta-block.stories.ts

```diff
@@ -0,0 +1,179 @@
+    linkWithIcon: {
+      href: 'https://www.example.com',
+      copy: 'Explore hybrid cloud infrastructure',
+    },
+  },
+  {
+    heading: 'Secure data and workloads',
+    copy: `Protect data beyond the platform and achieve regulatory compliance 
+    and resiliency with exceptional data security and privacy capabilities.`,
+    linkWithIcon: {
+      href: 'https://www.example.com',
+      copy: 'Explore infrastructure security',
+    },
+  },
+  {
+    heading: 'Flexible deployment',
+    copy: `Build once, run anywhere with flexible open compute solutions that 
+    support multiple Linux distributions, plus industry-leading cloud-native deployment.`,
+    linkWithIcon: {
+      href: 'https://www.example.com',
+      copy: 'Explore scale-out servers',
+    },
+  },
+  {
+    heading: 'Suspendisse nec est efficitur',
+    copy: `Manage and orchestrate your hybrid multicloud infrastructure with 
+    the most secure, reliable and flexible servers available today.`,
+    linkWithIcon: {
+      href: 'https://www.example.com',
+      copy: 'Lorem ipsum dolar',
+    },
+  },
+  {
+    heading: 'Morbi eget placerat felis',
+    copy: `Manage and orchestrate your hybrid multicloud infrastructure with 
+    the most secure, reliable and flexible servers available today.`,
+    linkWithIcon: {
+      href: 'https://www.example.com',
+      copy: 'Lorem ipsum dolar',
+    },
+  },
+];
+
+export const Default = ({ parameters }) => {
+  const { sectionHeading, secondaryHeading, mediaType } = parameters?.props?.CtaBlock ?? {};
+  return html`
+    <dds-cta-block>
+      <dds-content-section-heading>${ifNonNull(sectionHeading)}</dds-content-section-heading>
+
+      <dds-content-group slot="group-heading">
+        <dds-content-group-heading>${ifNonNull(secondaryHeading)}</dds-content-group-heading>
+        <dds-content-group-copy
+          >Lorem ipsum dolor sit amet, consecteture adipiscing elit sed dose. eiusmode tempor incididunt
+          ut.</dds-content-group-copy
+        >
+        <dds-link-with-icon slot="footer" href="${ifNonNull(hrefDefault)}">
+          Optional text link example ${ArrowRight20({ slot: 'icon' })}
+        </dds-link-with-icon>
+      </dds-content-group>
+
+      ${ctaBlockItems.map(
+        ({ heading: itemHeading, copy: itemCopy, linkWithIcon }) => html`
+          <dds-cta-block-item>
+            ${mediaType === 'image' ? image : ``} ${mediaType === 'video' ? video : ``}
+            ${mediaType === 'pictogram' ? pictogram : ``} ${mediaType === 'statistic' ? statistic : ``}
+            <dds-content-item-heading>${itemHeading}</dds-content-item-heading>
+            <dds-content-item-copy>${itemCopy}</dds-content-item-copy>
+            <dds-link-with-icon href="${linkWithIcon.href}" slot="footer">
+              ${linkWithIcon.copy} ${ArrowRight20({ slot: 'icon' })}
+            </dds-link-with-icon>
+          </dds-cta-block-item>
+        `
```

Comment:
I think render-from-data approach complicates the explanation of how to use the component, would you put plain markup here?